src/libraries/System.Private.CoreLib/src/System/Threading/ThreadPool.cs

```diff
@@ -594,7 +594,7 @@ internal static bool Dispatch()
                         return true;
                     }
 
-                    if (workQueue.loggingEnabled)
+                    if (FrameworkEventSource.Log.IsEnabled() && workQueue.loggingEnabled)
```

Comment:
May not make much difference but maybe it would be slightly better to swap the conditions:
```suggestion
                    if (workQueue.loggingEnabled && FrameworkEventSource.Log.IsEnabled())
```

When logging is not enabled it may be slightly cheaper to access the field (other things nearby are accessed from the same object), and when logging is enabled at verbose level the extra check wouldn't hurt. Hopefully ILLinker would generate the same code for both.